packages/appium/docs/en/commands/base-driver.md

```diff
@@ -319,9 +319,22 @@ Get data for all sessions running on an Appium server
 
 #### Response
 
-`MultiSessionData`<`C`\>[]
-
-A list of session data objects
+A list of session data objects.
```

Comment:
we should mark this endpoint as deprecated and not recommended for use, same as getSession one as they are not part of the W3C spec.

Reply:
Are there any replacements for those deprecated endpoints? They might be used in the Inspector.

Reply:
No for now I believe.
I think the inspector can be just "attach to a session" and users should give explicit session id. Current attach to an existing session does not let a user set an arbitrary session id by themselves, but it can have such setting an arbitrary session id to "attach to" the session.

Reply:
That would be a lot more inconvenient than the current process of automatic session discovery. Maybe this information can be added in the `/status` endpoint response, similarly to Selenium Grid? And for GET `session/:sessionId`, maybe something like GET `/session/:sessionId/appium/info`?
Can't comment on `getLog` and `getLogTypes`, but those could also just be moved to the corresponding `/session/:sessionId/appium/...` endpoint.

Reply:
Then, we might need to add `mobile: getSessions` as current base method to define additional commands not in w3c webdriver spec.

`/status` should be `ready` and `message` https://www.w3.org/TR/webdriver2/#status , so not for this usage.

Reply:
ah, or maybe we could define the `/sessions` endpoint in a new plugin? but not in the base driver?

Reply:
`mobile: getSessions` sort of implies it isn't compatible with other Appium-supported platforms like Mac. And while having this in a separate plugin is feasible, from a user perspective it would feel quite arbitrary to separate this.

I think `/status` could still work here because while it has to include `ready` and `message`, it "may additionally include arbitrary meta information that is specific to the implementation". [Selenium Grid also seems to do this](https://www.selenium.dev/documentation/grid/advanced_features/endpoints/#grid-status), though I'm not sure if its `/status` endpoint is W3C-compliant.

Reply:
then it looks like a new plugin for it would be nice. I don't know that response format by the grid, so we may need to check the response format first.
e.g. `selenium-grid-status` plugin for the new `status` format like the grid, `sessions-plugin` to keep `/sessions`

Reply:
https://github.com/appium/appium/issues/19856 for the plugin